libs/testdiff/testdiff.go

```diff
@@ -0,0 +1,90 @@
+package testdiff
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/databricks/cli/internal/testutil"
+	"github.com/hexops/gotextdiff"
+	"github.com/hexops/gotextdiff/myers"
+	"github.com/hexops/gotextdiff/span"
+	"github.com/stretchr/testify/assert"
+	"github.com/wI2L/jsondiff"
+)
+
+func UnifiedDiff(filename1, filename2, s1, s2 string) string {
+	edits := myers.ComputeEdits(span.URIFromPath(filename1), s1, s2)
+	return fmt.Sprint(gotextdiff.ToUnified(filename1, filename2, s1, edits))
+}
+
+func AssertEqualTexts(t testutil.TestingT, filename1, filename2, expected, out string) {
+	if len(out) < 1000 && len(expected) < 1000 {
+		// This shows full strings + diff which could be useful when debugging newlines
+		assert.Equal(t, expected, out)
+	} else {
+		// only show diff for large texts
+		diff := UnifiedDiff(filename1, filename2, expected, out)
+		t.Errorf("Diff:\n" + diff)
+	}
+}
+
+func AssertEqualJQ(t testutil.TestingT, expectedName, outName, expected, out string, ignorePaths []string) {
+	patch, err := jsondiff.CompareJSON([]byte(expected), []byte(out))
+	if err != nil {
+		t.Logf("CompareJSON error for %s vs %s: %s (fallback to textual comparison)", outName, expectedName, err)
+		AssertEqualTexts(t, expectedName, outName, expected, out)
+	} else {
+		diff := UnifiedDiff(expectedName, outName, expected, out)
+		t.Logf("Diff:\n%s", diff)
+		allowedDiffs := []string{}
+		erroredDiffs := []string{}
+		for _, op := range patch {
+			if allowDifference(ignorePaths, op) {
+				allowedDiffs = append(allowedDiffs, fmt.Sprintf("%7s %s %v old=%v", op.Type, op.Path, op.Value, op.OldValue))
+			} else {
+				erroredDiffs = append(erroredDiffs, fmt.Sprintf("%7s %s %v old=%v", op.Type, op.Path, op.Value, op.OldValue))
+			}
+		}
+		if len(allowedDiffs) > 0 {
+			t.Logf("Allowed differences between %s and %s:\n ==> %s", expectedName, outName, strings.Join(allowedDiffs, "\n ==> "))
+		}
+		if len(erroredDiffs) > 0 {
+			t.Errorf("Unexpected differences between %s and %s:\n ==> %s", expectedName, outName, strings.Join(erroredDiffs, "\n ==> "))
+		}
+	}
+}
+
+func allowDifference(ignorePaths []string, op jsondiff.Operation) bool {
+	if matchesPrefixes(ignorePaths, op.Path) {
+		return true
+	}
+	if op.Type == "replace" && almostSameStrings(op.OldValue, op.Value) {
+		return true
+	}
+	return false
+}
+
+// compare strings and ignore forward vs backward slashes
+func almostSameStrings(v1, v2 any) bool {
```

Comment:
Where is this needed? Didn't see it before.

Reply:
https://github.com/databricks/cli/pull/2025/commits/b972bcd50aec077b25ad6bde363dba82cc2997bc

This is to fix slashes on windows.  See below "whl" for example. https://github.com/databricks-eng/eng-dev-ecosystem/actions/runs/12432516778/job/34712095471

 ```
    testdiff.go:49: Allowed differences between C:\a\eng-dev-ecosystem\eng-dev-ecosystem\ext\cli\integration\bundle\testdata\default_python\bundle_summary.txt and Output from [bundle summary --output json]:
           ==> replace /bundle/terraform/exec_path C:\Users\RUNNER~1\AppData\Local\Temp\TestDefaultPython3.<NUMID>\<NUMID>\project_name_$UNIQUE_PRJ\.databricks\bundle\dev\bin\terraform.exe old=/tmp/.../terraform
           ==> replace /include/0 resources\project_name_$UNIQUE_PRJ.job.yml old=resources/project_name_$UNIQUE_PRJ.job.yml
           ==> replace /include/1 resources\project_name_$UNIQUE_PRJ.pipeline.yml old=resources/project_name_$UNIQUE_PRJ.pipeline.yml
           ==>  remove /resources/jobs/project_name_$UNIQUE_PRJ_job/email_notifications <nil> old=map[on_failure:[$USERNAME]]
           ==> replace /resources/jobs/project_name_$UNIQUE_PRJ_job/job_clusters/0/new_cluster/node_type_id Standard_D3_v2 old=i3.xlarge
           ==> replace /resources/jobs/project_name_$UNIQUE_PRJ_job/tasks/2/libraries/0/whl dist\*.whl old=dist/*.whl
           ==> replace /resources/jobs/project_name_$UNIQUE_PRJ_job/url https://$DATABRICKS_HOST/jobs/<NUMID> old=https://$DATABRICKS_HOST/jobs/<NUMID>?o=<NUMID>
           ==>  remove /resources/pipelines/project_name_$UNIQUE_PRJ_pipeline/catalog <nil> old=main
           ==> replace /resources/pipelines/project_name_$UNIQUE_PRJ_pipeline/url https://$DATABRICKS_HOST/pipelines/<UUID> old=https://$DATABRICKS_HOST/pipelines/<UUID>?o=<NUMID>
           ==>     add /workspace/current_user/externalId <UUID> old=<nil>
           ==> replace /workspace/current_user/groups/0/display admins old=team.engineering
           ==>  remove /workspace/current_user/name/familyName <nil> old=$USERNAME
```